lib/spack/spack/cmd/env.py

```diff
@@ -103,7 +103,7 @@ def env_activate(args):
         tty.die("No such environment: '%s'" % env)
 
     if spack_env == os.environ.get('SPACK_ENV'):
-        tty.die("Environment %s is already active" % args.activate_env)
```

Comment:
Maybe we should make this a regular or debug level message and return?

Reply:
I don't see why, this is the behavior when switching env:

```
$ spack env activate a 
$ spack env activate b
$ spack env activate a 
```

No output there

Reply:
I think people wouldn't activate the same environment twice on purpose in most cases, so my thought is that it would be good to leave a trace that this is happening.

Reply:
Also, with a debug level message you have the same as above unless you `spack -d env activate`

Reply:
If it's already unlikely people activate the same env twice, it's even more unlikely they do that with a `-d` flag, so I don't see the value of that to be honest.

Reply:
added it anyways